client/src/components/Onboarding/OnboardingUser.js

```diff
@@ -0,0 +1,259 @@
+  const username = params.username || authUsername;
+
+  if (hasSendingError) {
+    return (
+      <Message type="warn" className="appDetail-error">
+        We were unable to perform this action.
+      </Message>
+    );
+  }
+
+  return (
+    <span className="onboarding-actions">
+      {isStaff && step.state === 'staffwait' ? (
+        <span>
+          <Button
+            color="link"
+            disabled={isSending}
+            onClick={() => actionCallback('staff_approve', username)}
+          >
+            <h6>{step.staffApprove}</h6>
+          </Button>
+          <Button
+            color="link"
+            disabled={isSending}
+            onClick={() => actionCallback('staff_deny', username)}
+          >
+            <h6>{step.staffDeny}</h6>
+          </Button>
+        </span>
+      ) : null}
+      {step.state === 'userwait' ? (
+        <Button
+          color="link"
+          disabled={isSending}
+          onClick={() => actionCallback('user_confirm', username)}
+        >
+          <h6>{step.userConfirm}</h6>
+        </Button>
+      ) : null}
+      {isSending ? <LoadingSpinner placement="inline" /> : null}
+    </span>
+  );
+}
+
+OnboardingActions.propTypes = {
+  step: stepPropType.isRequired
+};
+
+OnboardingActions.defaultProps = {};
+
+function OnboardingStatus({ step }) {
+  let type = '';
+  switch (step.state) {
+    case 'processing':
+    case 'pending':
+      type = 'normal';
+      break;
+    case 'failed':
+      type = 'danger';
+      break;
+    case 'staffwait':
+    case 'userwait':
+      type = 'alert';
+      break;
+    case 'completed':
+      type = 'success';
+      break;
+    default:
+      type = 'normal';
+  }
+  if ('customStatus' in step) {
+    return <Pill type={type}>{step.customStatus}</Pill>;
+  }
+  switch (step.state) {
+    case 'pending':
+      return <Pill type={type}>Preparing</Pill>;
+    case 'staffwait':
+      return <Pill type="normal">Waiting for Staff Approval</Pill>;
+    case 'userwait':
+      return null;
+    case 'failed':
+      return <Pill type={type}>Unsuccessful</Pill>;
+    case 'completed':
+      return <Pill type={type}>Completed</Pill>;
+    case 'processing':
+      return (
+        <span className="onboarding-processing">
+          <Pill type={type}>Processing</Pill>
+          <LoadingSpinner placement="inline" />
+        </span>
+      );
+    default:
+      return <span>{step.state}</span>;
+  }
+}
+
+OnboardingStatus.propTypes = {
+  step: stepPropType.isRequired
+};
+
+OnboardingStatus.defaultProps = {};
+
+function OnboardingStep({ step }) {
+  return (
+    <div className="onboarding-step">
+      <div className="onboarding-step__name">{step.displayName}</div>
+      <div
+        /* eslint-disable react/no-danger */
+        className="onboarding-step__description"
+        dangerouslySetInnerHTML={{ __html: step.description }}
+      />
+      <div>
```

Comment:
__Question__: Why this extra `<div>`?

*__Thoughts__: A `<div>` serves no semantic purpose other than grouping. A `<div>` with no `className` suggests no purpose for the grouping. If this `<div>` is merely for breaking something to a new line; then (A) that does not seem necessary because a `<div>` is above this, and (B) using `display: block` (or other values) in the CSS can do this without requiring an extra `<div>`.*